ocaml/tests/test_ha_vm_failover.ml

```diff
@@ -569,4 +681,813 @@ module ComputeMaxFailures = Generic.MakeStateful (struct
+    , (* Assert that no_breach_plan returns as expected *)
+      [(vm2_grp1, master); (vm1_grp1, slave2)]
+    )
+  ; (* Test 7: Two anti-affinity VMs belong to one group in slave1 to be evacuated, only 1 got planed *)
+    ( {
+        basic_pool with
+        master= {memory_total= gib 512L; name_label= master; vms= []}
+      ; slaves=
+          [
+            {
+              memory_total= gib 256L
+            ; name_label= slave1
+            ; vms=
+                [
+                  {
+                    basic_vm with
+                    memory= gib 120L
+                  ; vm_name_label= vm1_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 513L
+                  ; vm_name_label= vm2_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ]
+            }
+          ; {memory_total= gib 256L; name_label= slave2; vms= []}
+          ]
+      }
+    , (* Assert that spread_evenly_plan returns as expected *)
+      [(vm1_grp1, slave2)]
+    , (* Assert that no_breach_plan returns as expected *)
+      [(vm1_grp1, slave2)]
+    )
+  ; (* Test 8: 6 anti-affinity VMs belong to one group in slave1 to be evacuated *)
+    ( {
+        basic_pool with
+        master= {memory_total= gib 640L; name_label= master; vms= []}
+      ; slaves=
+          [
+            {
+              memory_total= gib 256L
+            ; name_label= slave1
+            ; vms=
+                [
+                  {
+                    basic_vm with
+                    memory= gib 120L
+                  ; vm_name_label= vm2_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 60L
+                  ; vm_name_label= vm1_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 400L
+                  ; vm_name_label= vm6_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 250L
+                  ; vm_name_label= vm4_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 260L
+                  ; vm_name_label= vm5_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
+                  }
+                ; {
+                    basic_vm with
+                    memory= gib 130L
+                  ; vm_name_label= vm3_grp1
+                  ; groups= [{name_label= grp1; placement= AntiAffinity}]
```

Comment:
What about assigning a variable to `[{name_label= grp1; placement= AntiAffinity}]`?